src/Commands/Images/GenerateAbstract.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * 2019-present Friends of Presta community
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * that is bundled with this package in the file LICENSE
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/MIT
+ *
+ * @author Friends of Presta community
+ * @copyright 2019-present Friends of Presta community
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace FOP\Console\Commands\Images;
+
+use Configuration;
+use FOP\Console\Command;
+use Image;
+use ImageManager;
+use ImageType;
+use Language;
+use Module;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class GenerateAbstract extends Command
+{
+    /** @var array */
+    protected $errors = [];
+
+    /** @var OutputInterface */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+
+        return parent::initialize($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:images:generate:' . static::IMAGE_TYPE)
+            ->setDescription('Regenerate ' . static::IMAGE_TYPE . ' thumbnails')
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'images formats comma separated',
+                'all'
+            )
+            ->addOption(
```

Comment:
Nice catch that's because the default behavior in Prestashop controller is to delete the files.
Should we keep this behavior and just change the option name ?

Reply:
yep, that's ps admin option.

`--force` makes more sense to me.
Regenerate all files is a destructive operation and can be heavy,
Generating only missing images is lighter (usually) and not destructive.
Having the destructive behavior by adding an option makes sense.
I think it's safer this way.

Don't you ?

Reply:
The --force make more sense to me too but what about updated images ? will thumbnails be re generated ? Is there a check on the last modified date ?
Aka : I replace a product image by another one, do I need to --force to have the right thumbnails ?

Reply:
@jf-viguier nope the refresh of existing images is not managed by the script.

This is why the default behavior is to remove all existing images before generating the new ones. ( And the current default action of the command is the same )

```
->addOption(
                'delete',
                'd',
                InputOption::VALUE_OPTIONAL,
                'Delete current thumbnails',
                true
            );
```
Should i change the default behavior to keep the generated images and delete them only with the --force option ?

```
 ->addOption(
                'force',
                'f,
                InputOption::VALUE_OPTIONAL,
                'Delete current thumbnails',
                false
            );
```

Reply:
> Should i change the default behavior to keep the generated images and delete them only with the --force option ?
That's my preference.

Reply:
ok I understand why the default behaviour is to delete : we should do it by default.
In a future version, we could implement last modification date check.
So I understand that the delete=false means that I only want new images to be generated. So it could be more relevant to name it` --onlynew=1` or something. What do you think ?

Reply:
Another idea : if process failed, it could be useful to restart it with the --delete=0